docs/coding/javascript.md

```diff
@@ -78,89 +79,72 @@ do {
 ```
 These should all be followed by a semi-colon. In addition the `return` value expression must start on the same line as the return keyword in order to avoid semi-colon insertion.
 
-If the "Optimize JavaScript files" performance option in Drupal 6 is enabled, and there are missing semi-colons, then the JS aggregation will fail. It is therefore very important that semi-colons are used.
+- :heavy_check_mark: DO use a semi-colon except where it's not allowed
+- :heavy_check_mark: DO configure ESLint rule `semi` with option `["error", "always"]`.
```

Comment:
iirc, TypeScript needs to set this rule to `off` to work properly. Since you've added ESLint recommendations, do you think we should always add related TS rules?

``` json
"@typescript-eslint/semi": ["error", "always"],
"semi": "off",
```

Reply:
I wasn't sure if we wanted to have a dedicated set of guidelines for Typescript or not. I'll add a note about this nuance for now.